### PR TITLE
feat(web-client): bookmark

### PR DESCRIPTION
**Pull request by jongbonga**

### CODE REVIEW

File: web-client/src/app/state/bookmark/bookmark.service.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+@Injectable({ providedIn: 'root' })
+export class BookmarkService {
+  constructor(
+    private bookmarkStore: BookmarkStore,
+    private sessionQuery: SessionQuery,
+    private toastCtrl: ToastController
+  ) {}
+
+  async createBookmark(bookmark: { name: any; address: any }) {
```
Comment (jdvlio) [inline web-client/src/app/state/bookmark/bookmark.service.ts:21]: Could these instances of the `any` type be replaced with `string` instead?

**Review by jdvlio — CHANGES_REQUESTED** [review]

File: web-client/src/app/state/bookmark/bookmark.service.ts (hunk 1 above)
Comment (jongbonga) [reply web-client/src/app/state/bookmark/bookmark.service.ts]: @jdvlio done

**Review by sonasi — COMMENTED** [review]
image

Could we display a message in small text if there the bookmark list is empty, "no bookmarks", or whatever?

File: web-client/src/app/components/bookmark-list/bookmark-list.component.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            <ion-icon slot="icon-only" name="trash" color="danger"></ion-icon>
+          </ion-button>
+        </ion-buttons>
+      </ion-item>
+    </ng-container>
+  </div>
+
+  <ng-template #empty>
+    <div class="ion-padding-horizontal ion-text-center">
+      <p>You have not added a recipient yet</p>
```
Comment (jongbonga) [inline web-client/src/app/components/bookmark-list/bookmark-list.component.html:30]: @sonasi I have a message here, I need to debug the address you are using

File: web-client/src/app/components/bookmark-list/bookmark-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(
+    public bookmarkQuery: BookmarkQuery,
+    private bookmarkService: BookmarkService,
+    private navCtrl: NavController
+  ) {}
+
+  ngOnInit() {}
+
+  sendFund(receiverAddress: string) {
+    this.navCtrl.navigateForward('pay', {
```
[the remark this answers is not in the document]
Comment (jongbonga) [reply web-client/src/app/components/bookmark-list/bookmark-list.component.ts:22]: @sonasi this is the pay page, it verifies the address to send to, what address did you use for the bookmark?

**Comment by billguo99** [comment]
I launched ntls-staging environment locally and tested a wallet. (r3akVmwDcDjqPtsjbFbNkGH5dEAmMLn6EW)
I clicked on Bookmark Recipients and got the following errors in the console:
image

It doesn't interfere with the application, but are these console errors expected?

File: web-client/src/app/components/bookmark-list/bookmark-list.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (billguo99) [reply web-client/src/app/components/bookmark-list/bookmark-list.component.ts:22]: I get the same "Address not recognised" - I bookmarked these two addresses:
```
Ohne Pixelc: r3akVmwDcDjqPtsjbFbNkGH5dEAmMLn6EW
Ohne Pixeld: rHbhPzLaXhjqJKzLdsADKADgwPCQDWHt3L
```

[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/components/bookmark-list/bookmark-list.component.ts:22]: I am also getting the same error using the following address `rhDanuXviGtZB7Zo3NNN5Qozx7ssY4nVCe`

Plus, the back button does not to appear when the `Unsupported Address` page appears.

**Review by sonasi — APPROVED** [review]

**Review by IscoRuta98 — COMMENTED** [review]
LGTM!
I was able to: save & delete bookmark, and send funds via the icon cash. 

One design suggestion is perhaps adding an error message when someone tries to  save an address that does not meet the required minimum number of characters. Something similar to a message that appears when the PIN does not meet the requirement.

image

**Review by sonasi — APPROVED** [review]

**Review by sonasi — APPROVED** [review]

**Review by jdvlio — APPROVED** [review]

**Pull request merged**